ci/scripts/cpp_build.sh

```diff
@@ -118,12 +118,20 @@ if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
     fi
   }
 
+  ORIGINAL_CXX=${CXX}
+  if [ "${ARROW_USE_CCACHE}" = "ON" ]; then
+      export CXX="ccache ${CXX}"
+  else
+      export CXX="sccache ${CXX}"
```

Comment:
```suggestion
    if command -v sccache &> /dev/null; then
      export CXX="sccache ${CXX}"
    fi
```

Reply:
command -v is guaranteed to not produce stderr, so you only need to redirect stdout.